search-api/src/main/scala/no/ndla/searchapi/integration/TaxonomyApiClient.scala

```diff
@@ -34,63 +34,93 @@ trait TaxonomyApiClient {
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      ).map(_.distinct)
 
-    def getAllTopicResourceConnections: Try[List[TopicResourceConnection]] =
-      get[List[TopicResourceConnection]](s"$TaxonomyApiEndpoint/topic-resources/")
-
-    def getAllTopicSubtopicConnections: Try[List[TopicSubtopicConnection]] =
-      get[List[TopicSubtopicConnection]](s"$TaxonomyApiEndpoint/topic-subtopics/")
+    def getAllTopicResourceConnections(shouldUsePublishedTax: Boolean): Try[List[TopicResourceConnection]] =
+      get[List[TopicResourceConnection]](
+        s"$TaxonomyApiEndpoint/topic-resources/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      )
 
-    def getAllResourceResourceTypeConnections: Try[List[ResourceResourceTypeConnection]] =
-      get[List[ResourceResourceTypeConnection]](s"$TaxonomyApiEndpoint/resource-resourcetypes/").map(_.distinct)
+    def getAllTopicSubtopicConnections(shouldUsePublishedTax: Boolean): Try[List[TopicSubtopicConnection]] =
+      get[List[TopicSubtopicConnection]](
+        s"$TaxonomyApiEndpoint/topic-subtopics/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      )
 
-    def getAllSubjectTopicConnections: Try[List[SubjectTopicConnection]] =
-      get[List[SubjectTopicConnection]](s"$TaxonomyApiEndpoint/subject-topics/")
+    def getAllResourceResourceTypeConnections(
+        shouldUsePublishedTax: Boolean
+    ): Try[List[ResourceResourceTypeConnection]] =
+      get[List[ResourceResourceTypeConnection]](
+        s"$TaxonomyApiEndpoint/resource-resourcetypes/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      ).map(_.distinct)
+
+    def getAllSubjectTopicConnections(shouldUsePublishedTax: Boolean): Try[List[SubjectTopicConnection]] =
+      get[List[SubjectTopicConnection]](
+        s"$TaxonomyApiEndpoint/subject-topics/",
+        headers = getVersionHashHeader(shouldUsePublishedTax)
+      )
 
-    def getAllRelevances: Try[List[Relevance]] =
-      get[List[Relevance]](s"$TaxonomyApiEndpoint/relevances/").map(_.distinct)
+    def getAllRelevances(shouldUsePublishedTax: Boolean): Try[List[Relevance]] =
+      get[List[Relevance]](s"$TaxonomyApiEndpoint/relevances/", headers = getVersionHashHeader(shouldUsePublishedTax))
+        .map(_.distinct)
 
-    def getSearchableTaxonomy(contentUri: String, filterVisibles: Boolean): Try[List[SearchableTaxonomyContext]] = {
+    def getSearchableTaxonomy(
+        contentUri: String,
+        filterVisibles: Boolean,
+        shouldUsePublishedTax: Boolean
+    ): Try[List[SearchableTaxonomyContext]] = {
       implicit val formats = SearchableLanguageFormats.JSonFormatsWithMillis
       get[List[SearchableTaxonomyContext]](
         s"$TaxonomyApiEndpoint/queries/$contentUri",
-        "filterVisibles" -> filterVisibles.toString
+        headers = getVersionHashHeader(shouldUsePublishedTax) + Map("filterVisibles" -> filterVisibles.toString)
```

Comment:
Må ha dobbel + for å slå sammen maps.
```suggestion
        headers = getVersionHashHeader(shouldUsePublishedTax) ++ Map("filterVisibles" -> filterVisibles.toString)
```